### PR TITLE
Vendor gardener/gardener@v1.40.2

### PR DESCRIPTION
**Pull request by jia-jerry**
**How to categorize this PR?**

/kind bug
/platform alicloud

**What this PR does / why we need it**:
Regular vendor g/g and fix some critical issues like https://github.com/gardener/gardener/pull/5474
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```breaking user
kubernetes v1.15 and v1.16 will not be supported.
```

### CODE REVIEW

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by dkistner** [comment]
Are you not going to adapt to use `ginkgo@v2`?

**Comment by jia-jerry** [comment]
> Are you not going to adapt to use `ginkgo@v2`?

Will open another PR to adopt it.

**Review by rfranzke — COMMENTED** [review]
Can you update https://github.com/gardener/gardener-extension-provider-alicloud/#supported-kubernetes-versions as well?

**Comment by jia-jerry** [comment]
> Can you update https://github.com/gardener/gardener-extension-provider-alicloud/#supported-kubernetes-versions as well?

done.

File: README.md [hunk 1]
```diff
@@ -31,9 +31,6 @@ This extension controller supports the following Kubernetes versions:
 | Kubernetes 1.18 | 1.18.0+     | [![Gardener v1.18 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.18%20Alibaba%20Cloud/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.18%20Alibaba%20Cloud) |
 | Kubernetes 1.17 | 1.17.0+     | [![Gardener v1.17 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.17%20Alibaba%20Cloud/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.17%20Alibaba%20Cloud) |
 | Kubernetes 1.16 | 1.16.0+     | [![Gardener v1.16 Conformance Tests](https://testgrid.k8s.io/q/summary/conformance-gardener/Gardener,%20v1.16%20Alibaba%20Cloud/tests_status?style=svg)](https://testgrid.k8s.io/conformance-gardener#Gardener,%20v1.16%20Alibaba%20Cloud) |
```
Comment (rfranzke) [inline README.md:33]: Also drop 1.16

Comment (jia-jerry) [reply README.md]: done.

**Review by rfranzke — APPROVED** [review]

**Pull request merged**